### PR TITLE
Include MTU option for KuryrConfig

### PR DESCRIPTION
**Pull request by MaysaMacedo**
When using a Provider Network for the Machines
the same MTU is attempted to be used for the
Tenant Networks of the Pods, which can be considered
too big by Neutron. A new option is being included
to allow the user to set the desired MTU for the Pods
Network to avoid such scenario.

### CODE REVIEW

File: operator/v1/types_network.go [hunk 1]
```diff
@@ -295,6 +295,13 @@ type KuryrConfig struct {
 	// +optional
 	EnablePortPoolsPrepopulation bool `json:"enablePortPoolsPrepopulation,omitempty"`
 
+	// mtu is the MTU to use for the Pods Network.
+	// This must be smaller than the Nodes Network MTU.
+	// If unset the Nodes Network MTU is used.
+	// +kubebuilder:validation:Minimum=0
+	// +optional
+	MTU *uint32 `json:"mtu,omitempty"`
```
Comment (dulek) [inline operator/v1/types_network.go:303]: Does `nil` value make sense from our perspective? We could treat `0` as the default value, right? I see it's trying to be consistent with other *SDNConfig structs, so I guess I'm okay with this.

File: operator/v1/0000_70_cluster-network-operator_01_crd.yaml [hunk 2]
```diff
@@ -207,6 +207,13 @@ spec:
                           Kuryr keeps a number of ports ready to be attached to pods.
                           By default port prepopulation is disabled.
                         type: boolean
+                      mtu:
+                        description: mtu is the MTU to use for the Pods Network.
+                          This must be smaller than the Nodes Network MTU.
+                          If unset, defaults to the Nodes Network MTU.
```
Comment (dulek) [inline operator/v1/0000_70_cluster-network-operator_01_crd.yaml:213]: I'd rephrase this to something like this:

```
mtu is the MTU that Kuryr should use when creating pod networks in Neutron. The value has to be lower or equal to the MTU of the nodes network and Neutron has to allow creation of tenant networks with such MTU. If unset Pod networks will be created with the same MTU as the nodes network has.
```

File: operator/v1/types_network.go [hunk 3]
```diff
@@ -295,6 +295,13 @@ type KuryrConfig struct {
 	// +optional
 	EnablePortPoolsPrepopulation bool `json:"enablePortPoolsPrepopulation,omitempty"`
 
+	// mtu is the MTU to use for the Pods Network.
+	// This must be smaller than the Nodes Network MTU.
+	// If unset the Nodes Network MTU is used.
```
Comment (dulek) [inline operator/v1/types_network.go:300]: Same here.

**Review by dulek — CHANGES_REQUESTED** [review]

File: operator/v1/types_network.go (hunk 1 above)
Comment (MaysaMacedo) [reply operator/v1/types_network.go:304]: yup, I believe it makes sense to be an optional setting, even when Provider Network is used. If issues are hit the user will have the ability to configure it with this setting.

File: operator/v1/types_network.go [hunk 4]
```diff
@@ -318,6 +318,14 @@ type KuryrConfig struct {
 	// +kubebuilder:validation:Minimum=0
 	// +optional
 	PoolBatchPorts *uint `json:"poolBatchPorts,omitempty"`
+
+	// mtu is the MTU that Kuryr should use when creating pod networks in Neutron.
+	// The value has to be lower or equal to the MTU of the nodes network and Neutron has
+	// to allow creation of tenant networks with such MTU. If unset Pod networks will be
+	// created with the same MTU as the nodes network has.
+	// +kubebuilder:validation:Minimum=0
```
Comment (deads2k) [inline operator/v1/types_network.go:326]: what happens if you actually specify something like one.  That doesn't seem likely to do good things.

**Comment by deads2k** [comment]
This describes the defaults and is consistent with other network types.

/approve

I'll leave lgtm up to someone more familiar with networking.

**Comment by deads2k** [comment]
I'm am compelled to ask. Why are the mtu default values different for all these network provider types?  It seems like present us would benefit from keeping these things the same.

**Comment by MaysaMacedo** [comment]
> I'm am compelled to ask. Why are the mtu default values different for all these network provider types? It seems like present us would benefit from keeping these things the same.

Those networks use different tunnelling mechanisms. Depending on the tunnelling used the value reserved for the header may differ, consequently the value of MTU varies to make room for that header.

File: operator/v1/types_network.go (hunk 4 above)
Comment (MaysaMacedo) [reply operator/v1/types_network.go:326]: Even though there is little sense, Neutron allows to create Networks/Ports with MTU 1.
But we can always try to enforce the minimum value allowed on the cluster-network-operator as other SDNs do [1].

[1] https://github.com/openshift/cluster-network-operator/blob/master/pkg/network/openshift_sdn.go#L110-L112
